File: create-target-array-in-the-given-order/create-target-array-in-the-given-order.js
/**
 * @param {number[]} nums
 * @param {number[]} index
 * @return {number[]}
 */
var createTargetArray = function(nums, index) {    
  return nums.reduce((p, c, i) => {
      p.splice(index[i], 0, c);
      return p;
  }, []);
};

File: create-target-array-in-the-given-order/create-target-array-in-the-given-order.go
func createTargetArray(nums []int, index []int) []int {
	result := []int{}
	for i := range nums {
		temp := append([]int{}, result[:index[i]]...)
		temp = append(temp, nums[i])
		temp = append(temp, result[index[i]:]...)
		result = temp
	}
	return result
}